crates/app/src/lib.rs

```diff
@@ -353,6 +368,14 @@ impl<'a> AppTrigger<'a> {
     }
 }
 
+// "Seal" MaybeLoader so external crates cannot implement it
+mod private {
+    pub trait MaybeLoader {}
+}
+use private::MaybeLoader;
+impl MaybeLoader for AppLoader {}
+impl MaybeLoader for () {}
```

Comment:
Nit: if we do go this route, it might be better to make a `struct InertLoader;` instead of doing this on unit which is not very self-describing